### PR TITLE
*: limit request size for v3

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/1978

/cc @heyitsanthony @gyuho

### CODE REVIEW

File: Documentation/rfc/v3api.md [hunk 1]
```diff
@@ -34,6 +34,15 @@ To prove out the design of the v3 API the team has also built [a number of examp
     - easy for people to write simple etcd application
 
 
+## Notes
+
+### Request Size Limitation
+
+The max request size is around 1MB. etcd replicates requests in a streaming fashion, a very large
+request might block other requests from a long time. The use case for etcd is to store small configuration
+values, so we avoid user from submitting large requests. This also applies to Txn requests. We might loose
```
Comment (heyitsanthony) [inline Documentation/rfc/v3api.md]: "loosen"?

File: Documentation/rfc/v3api.md [hunk 2]
```diff
@@ -34,6 +34,15 @@ To prove out the design of the v3 API the team has also built [a number of examp
     - easy for people to write simple etcd application
 
 
+## Notes
+
+### Request Size Limitation
+
+The max request size is around 1MB. etcd replicates requests in a streaming fashion, a very large
+request might block other requests from a long time. The use case for etcd is to store small configuration
```
Comment (heyitsanthony) [inline Documentation/rfc/v3api.md]: "for a long time"?

**Comment by gyuho** [comment]
lgtm.

File: Documentation/rfc/v3api.md (hunk 1 above)
Comment (heyitsanthony) [inline Documentation/rfc/v3api.md]: "we prevent the user"?

File: Documentation/rfc/v3api.md [hunk 3]
```diff
@@ -34,6 +34,15 @@ To prove out the design of the v3 API the team has also built [a number of examp
     - easy for people to write simple etcd application
 
 
+## Notes
+
+### Request Size Limitation
+
+The max request size is around 1MB. etcd replicates requests in a streaming fashion, a very large
```
Comment (heyitsanthony) [inline Documentation/rfc/v3api.md]: "Since etcd replicates"?

File: etcdserver/v3demo_server.go [hunk 4]
```diff
@@ -29,6 +29,14 @@ import (
 	"github.com/coreos/etcd/storage/storagepb"
 )
 
+const (
+	// the max request size that raft accepts.
+	// TODO: make this a flag? But we probably do not want to
+	// accept large request which might block raft stream. User
+	// specify a large value might end up with shooting in the foot.
+	maxRequestSize = 1.5 * 1024 * 1024
```
Comment (heyitsanthony) [inline etcdserver/v3demo_server.go]: `maxRequestBytes`?

**Comment by heyitsanthony** [comment]
lgtm aside from minor wording nits

**Pull request merged**